### PR TITLE
Check feed suggestion after values are disputed

### LINKED ISSUE #669: Check for Disputed Value Tip availability
When a reporter submits a value that qualifies for a tip, that tip is made available again if the submitted value is disputed. Currently the telliot tip listener may not be hearing these tips. This issue wants to make sure that the data keeps flowing in these situations.

### PR DESCRIPTION
**Pull request by akremstudy**
Closes #669 
This pr was mainly to check if a feed will be suggested if a tip eligible submission is disputed.
Added a test to mimic that scenario.
In addition, replaced  the mock tellor contracts with full contracts, just to remove doubt of any errors while testing that may be due to the contracts being mocks.
Had to edit some tests to use updated contracts.
### Steps Taken to QA Changes

- Ran all the tests
### Checklist

This pull request is:

- [ ] A documentation error, docs update, or typographical error fix
	- No tests or issue needed
- [ ] A code fix
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. Fixes without tests will not be accepted unless it's related to the documentation only.
    - Please make sure docs are updated if need be
- [ ] A feature implementation
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure docs updates are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -138,56 +152,70 @@ def local_node_cfg(chain_id: int):
...
     """mock oracle(TellorFlex) contract to stake in"""
-    return accounts[0].deploy(TellorFlex, mock_token_contract.address, accounts[0], 10e18, 60)
+    return accounts[0].deploy(
+        TellorFlex,
+        mock_token_contract.address,
+        43200,
+        Web3.toWei(100, "ether"),
+        Web3.toWei(15, "ether"),
+        Web3.toWei(10, "ether"),
+        HexBytes("0x5c13cd9c97dbb98f2429c101a2a8150e6c7a0ddaff6124ee176a3a411067ded0"),
```
Comment (oraclown) [inline tests/conftest.py:191]: What's this? Add comment?

File: tests/tips/test_tips_disputed_submissions.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+def deploy(query_data_storage_contract):
+    # deploy playground
+    playground = accounts[0].deploy(TellorPlayground)
+    # deploy autopay
+    autopay = accounts[0].deploy(Autopay, playground.address, query_data_storage_contract.address, 20)
+    return playground, autopay
+
+
+@pytest.mark.asyncio
+async def test_tips_disputed_submissions(deploy, mumbai_test_cfg):
```
Comment (oraclown) [inline tests/tips/test_tips_disputed_submissions.py:23]: This is a great test addition, but I think you could make what's happening clearer for future readers/maintainers/contributors. The comments within the test function are helpful. Perhaps add a docstring explaining the overall flow and purpose of the test, similar to your explanation in the PR description.

File: tests/conftest.py [hunk 3]
```diff
@@ -33,6 +35,18 @@ def shared_setup(module_isolation):
     pass
 
 
+@pytest.fixture(scope="module", autouse=True)
+def reset_chain(chain):
```
Comment (oraclown) [inline tests/conftest.py:39]: What's this used for? I didn't see it used in your other code changes, but maybe I missed it.

File: tests/conftest.py [hunk 4]
```diff
@@ -33,6 +35,18 @@ def shared_setup(module_isolation):
...
+    chain.reset()
+
+
+@pytest.fixture(scope="function", autouse=True)
+def snapshot(chain):
```
Comment (oraclown) [inline tests/conftest.py:44]: What's this used for? I didn't see it used in your other code changes, but maybe I missed it.

**Review by oraclown — APPROVED** [review]
Thanks for this, @akremstudy ! Opened #678 as a possible future improvement, and am interested in what you think. Great test addition, and nice updates using the actual contracts. I left some questions / suggestions.

File: tests/conftest.py (hunk 3 above)
Comment (akremstudy) [reply tests/conftest.py:39]: resets the chain after each module

File: tests/conftest.py (hunk 4 above)
Comment (akremstudy) [reply tests/conftest.py:45]: resets the chain after each function

**Pull request merged**